federation/pkg/federation-controller/sync/controller.go

```diff
@@ -438,10 +450,32 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		return statusNotSynced
 	}
 
-	operations, err := operationsAccessor(adapter, clusters, obj)
+	var userInfo interface{}
+	if objUpdatingAdapter, ok := adapter.(federatedtypes.ObjectUpdatingAdapter); ok {
+		userInfo, err = objUpdatingAdapter.PrepareForUpdate(obj, key, clusters, informer)
```

Comment:
I believe that my work in #46252 will address this: the cluster selection happens before any of the scheduling, which is what I believe is supposed to happen.